src/Microsoft.Framework.ApplicationHost/Impl/Syntax/CommandGrammar.cs

```diff
@@ -9,12 +9,12 @@ namespace Microsoft.Framework.ApplicationHost.Impl.Syntax
 {
     internal class CommandGrammar : Grammar
     {
-        public CommandGrammar(Func<string, string> variable)
+        private CommandGrammar(Func<string, string> variable, bool preserveSurroundingQuotes)
         {
             var environmentVariablePiece = Ch('%').And(Rep(Ch().Not(Ch('%')))).And(Ch('%')).Left().Down().Str()
                 .Build(key => variable(key) ?? "%" + key + "%");
 
-            var escapeSequencePiece = 
+            var escapeSequencePiece =
                 Ch('%').And(Ch('%')).Build(_=>"%")
                     .Or(Ch('^').And(Ch('^')).Build(_ => "^"))
                     .Or(Ch('\\').And(Ch('\\')).Build(_ => "\\"))
```

Comment:
Didn't change `escapeSequencePiece` but it probably should not remove Windows (`'^'`) or Linux (`'\\'`) escapes when `preserveSurroundingQuotes`. Should be fine for now since double quotes are far more common than escaping a single character.